feat: Write unit tests for iFrame (STARGATE-1695)

lukewhyte opened the pull request:
The most difficult tests are completed (nested mock api calls). Just the easy tests to add.

lukewhyte committed:
feat: Write unit tests for iFrame (STARGATE-1695)

lukewhyte commented on packages/pci-sdk-iframe/src/pages/app/useApp.ts:
```diff
@@ -40,20 +40,22 @@ export default function useApp() {
-				default:
-					break;
-			}
+			try {
```
This needs to be fixed. In the tests `window.parent.postMessage('apto-iframe-ready', '*')` is caught by `onMessage` and, since it isn't JSON, JSON.parse throws an error.

IagoLast reviewed (commented):
I need to take a deeper look at this.

Maybe we want to use msw here too istead reusing the fetchStub logic

lukewhyte committed:
Refactoring existing tests

lukewhyte committed:
Tests complete

lukewhyte committed:
Updated packages

IagoLast committed:
chore: Update eslint

- This commit adds eslint/typescript as a dependency and fixes all the issues found on the codebase.
- Keep in mind we were using queryByRole and role for cardholder-name, thats not a valid aria role!
- For some reason eslint is not working fine on vscode (I guess some lerna interference)

IagoLast committed:
Fix eslint configuration

lukewhyte committed:
unit tests passing

lukewhyte committed:
Added private env variables for auth
